src/Utils/obiDialogParser.ts

```diff
@@ -456,9 +502,33 @@ export class ObiDialogParser {
             throw new Error(`Failed to create entity ${entityName}`)
         }
         newEntity.entityId = entityId
+        this.entities.push(newEntity)
         return newEntity
     }
 
+    /**
+     * Returns a `Condition` referencing an enum entity and enum value id, which is built from the entity and value name.
+     */
+    private getConditionFromRequiredEntity(requiredEntity: OBIUtils.ConditionEntityAndValue): CLM.Condition {
+        const conditionEntity = this.entities.find(e => e.entityName === requiredEntity.entity)
+        if (!conditionEntity) {
+            throw new Error(`Couldn't find entity ${requiredEntity.entity}`)  // Unexpected, shouldn't happen.
```

Comment:
nit: prefer comments on separate lines than code, or in other codes comments and code not on same lines. Also think the comment should be on the condition / `if` statement not the error throwing of the error if you're trying to say the condition occurring shouldn't happen not the throwing of the error.

Reply:
Moved to separate lines.  I'm keeping the comment inside the `if` block though; I think it's more clear that the comment is related to that condition.  It looks strange to my eye to put the comment before the `if`.